src/plugins/intel_gpu/src/plugin/ops/ctc_greedy_decoder.cpp

```diff
@@ -43,67 +43,115 @@ static void CreateCommonCTCGreedyDecoderOp(ProgramBuilder& p, const std::shared_
         }
     }
 
-    uint32_t blank_index = static_cast<uint32_t>(op->get_input_shape(0).back() - 1);
-    if (reordered_inputs.size() == 3) {
-        auto blank_index_node = std::dynamic_pointer_cast<ov::op::v0::Constant>(op->get_input_node_shared_ptr(2));
-        if (!blank_index_node) {
-            OPENVINO_THROW("Unsupported blank_index node type in ", op->get_friendly_name(), " (", op->get_type_name(), ")");
+    if (p.use_new_shape_infer()) {
+        size_t num_outputs = op->get_output_size();
+
+        auto get_output_paddings = [&]() {
+            std::vector<cldnn::padding> output_paddings;
+            for (size_t i = 0; i < num_outputs; i++)
+                output_paddings.push_back(cldnn::padding());
+            return output_paddings;
+        };
+
+        auto get_output_data_types = [&]() {
+            std::vector<cldnn::optional_data_type> output_data_types;
+            for (size_t i = 0; i < num_outputs; i++) {
+                auto type = op->get_output_element_type(i);
+                output_data_types.push_back(cldnn::element_type_to_data_type(type));
+            }
+            return output_data_types;
+        };
+
+        uint32_t blank_index = UINT32_MAX;
+        if (reordered_inputs.size() == 3) {
+            auto blank_index_node = std::dynamic_pointer_cast<ov::op::v0::Constant>(op->get_input_node_shared_ptr(2));
+            if (!blank_index_node) {
+                OPENVINO_THROW("Unsupported blank_index node type in ", op->get_friendly_name(), " (", op->get_type_name(), ")");
+            }
+            float val;
+            if (ov::shape_size(blank_index_node->get_output_shape(0)) != 1 || !ov::op::util::get_single_value(blank_index_node, val)) {
+                OPENVINO_THROW("Unsupported parameter size in ", op->get_friendly_name(), " (", op->get_type_name(), ")");
+            }
+            blank_index = static_cast<uint32_t>(val);
+            reordered_inputs.pop_back();
         }
-        float val;
-        if (ov::shape_size(blank_index_node->get_output_shape(0)) != 1 || !ov::op::util::get_single_value(blank_index_node, val)) {
-            OPENVINO_THROW("Unsupported parameter size in ", op->get_friendly_name(), " (", op->get_type_name(), ")");
+
+        auto primitive = cldnn::ctc_greedy_decoder(
+                    layer_type_name_ID(op),
+                    reordered_inputs,
+                    blank_index,
+                    ctc_merge_repeated,
+                    cldnn::padding({0, 0, 0, 0}, 0),
+                    cldnn::element_type_to_data_type(op->get_output_element_type(0)),
+                    op->get_output_size());
+        primitive.output_paddings = get_output_paddings();
+        primitive.output_data_types = get_output_data_types();
+        if (num_outputs != 1)
+            primitive.use_multiple_outputs = true;
```

Comment:
If use_multiple_outputs is checked only as not having one output, why is this used as a flag?
Is it available to check the num_outputs instead of use_multiple_outputs ?

Reply:
If you add it to the primitive, you should also add it to the method of load/save/hash methods of the primitive.

Reply:
> If use_multiple_outputs is checked only as not having one output, why is this used as a flag? Is it available to check the num_outputs instead of use_multiple_outputs ?

num_outputs is used for static case multiple outputs, and use_multiple_outputs is for dynamic case multiple outputs.

Reply:
> If you add it to the primitive, you should also add it to the method of load/save/hash methods of the primitive.

Yes, they were added. Thanks for your comment.

Reply:
>> num_outputs is used for static case multiple outputs, and use_multiple_outputs is for dynamic case multiple outputs.

I cannot understand why those are distinguished?  What is the difference?

Reply:
> > > num_outputs is used for static case multiple outputs, and use_multiple_outputs is for dynamic case multiple outputs.
> 
> I cannot understand why those are distinguished? What is the difference?

In CTCGreedyDecoder not CTCGreedyDecoderSeqLen, num_outputs is 1 in dynamic case.